src/main/java/net/minestom/server/entity/Entity.java

```diff
@@ -470,6 +470,56 @@ public final boolean removeViewer(@NotNull Player player) {
         return true;
     }
 
+    /**
+     * Creates a copy of this entity with all data.
+     * <p>
+     * Subclasses may have to override {@link #fillCopyWithData(Entity)} and {@link #createCopyInstance()} for correct saving.
+     * This is going to be used for saving/loading entities in the ChunkLoader.
+     * <p>
+     * The position data of the copy will be set, but not the chunk, considering it is not in a chunk anymore.
+     *
+     * @return a copy with all entity data
+     * @throws UnsupportedOperationException if a copy can't be created, for example, because this entity is a player
+     * @see #fillCopyWithData(Entity)
+     * @see #createCopyInstance()
+     */
+    public final Entity createCopy() throws UnsupportedOperationException {
+        Entity copy = createCopyInstance();
```

Comment:
It should probably be an error if the type of Entity is not the same exact type as `this`